### PR TITLE
Upgrade dependencies and CI config

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: test/bench_internal/parser_benchmarks.cpp [hunk 1]
```diff
@@ -100,4 +100,4 @@ static void parse_string(benchmark::State& state)
 
     state.SetItemsProcessed(static_cast<int64_t>(size));
 }
-BENCHMARK(parse_string)->RangeMultiplier(2)->Range(16, 4 * 1024);
+BENCHMARK(parse_string)->RangeMultiplier(2)->Range(16, 4096);
```
Comment (axic) [inline test/bench_internal/parser_benchmarks.cpp:103]: Is this change needed after `-bugprone-implicit-widening-of-multiplication-result` ?

File: test/utils/wasm3_engine.cpp [hunk 2]
```diff
@@ -124,7 +124,7 @@ WasmEngine::Result Wasm3Engine::execute(
 {
     unsigned ret_valid;
     uint64_t ret_value;
-    IM3Function function = reinterpret_cast<IM3Function>(func_ref);
+    auto function = reinterpret_cast<IM3Function>(func_ref);  // NOLINT(performance-no-int-to-ptr)
```
Comment (axic) [inline test/utils/wasm3_engine.cpp:127]: Why was this needed?

File: lib/fizzy/instantiate.cpp [hunk 3]
```diff
@@ -203,7 +203,7 @@ std::tuple<bytes_ptr, Limits> allocate_memory(const std::vector<Memory>& module_
         }
 
         // NOTE: fill it with zeroes
-        bytes_ptr memory{new bytes(memory_min * PageSize, 0), bytes_delete};
+        bytes_ptr memory{new bytes(uint32_t{memory_min * PageSize}, 0), bytes_delete};
```
Comment (axic) [inline lib/fizzy/instantiate.cpp:206]: @gumb0 can you double check?

File: CMakeLists.txt [hunk 4]
```diff
@@ -66,7 +66,6 @@ if(CMAKE_CXX_COMPILER_ID MATCHES Clang)
             -Wno-double-promotion
             -Wno-float-equal
             -Wno-padded
-            -Wno-return-std-move-in-c++11
```
Comment (axic) [inline CMakeLists.txt:69]: What was this causing?

File: test/bench_internal/parser_benchmarks.cpp (hunk 1 above)
Comment (chfast) [reply test/bench_internal/parser_benchmarks.cpp:103]: No.

File: test/utils/wasm3_engine.cpp (hunk 2 above)
Comment (chfast) [reply test/utils/wasm3_engine.cpp:127]: This is probably new clang-tidy warning (as I upgraded to clang-tidy-13). Pointer to integer cast blocks some optimizations. https://clang.llvm.org/extra/clang-tidy/checks/performance-no-int-to-ptr.html.

File: CMakeLists.txt (hunk 4 above)
Comment (chfast) [reply CMakeLists.txt:69]: This flag does not exist any more.

File: lib/fizzy/instantiate.cpp (hunk 3 above)
Comment (chfast) [reply lib/fizzy/instantiate.cpp:206]: Note that this is non-functional change. The `uint32_t{}` explicitly states that the multiplication is done with 32-bit precision. This will not compile if any of multiplication operands has bigger type.

File: test/bench_internal/parser_benchmarks.cpp (hunk 1 above)
Comment (gumb0) [reply test/bench_internal/parser_benchmarks.cpp:103]: why change then?

File: lib/fizzy/instantiate.cpp (hunk 3 above)
Comment (gumb0) [reply lib/fizzy/instantiate.cpp:206]: Is this change still needed with disabled `bugprone-implicit-widening-of-multiplication-result`?

Comment (chfast) [reply lib/fizzy/instantiate.cpp:206]: It is not. I started "fixing" these but it turned out there are too many cases for constant multiplication. I reported the bug about the check but it is not worth it in current form.

Comment (gumb0) [reply lib/fizzy/instantiate.cpp:206]: Well turns out we have a bug on master here, because trying to allocate 4Gb actually overflows and allocates 0.

https://github.com/wasmx/fizzy/pull/805

File: .codespell-whitelist [hunk 5]
```diff
@@ -1 +1,2 @@
 mut
+crate
```
Comment (gumb0) [inline .codespell-whitelist:2]: Also not sure why, it's a normal English word?

Comment (chfast) [reply .codespell-whitelist:2]: I don't know. The codespell suggests crate -> create.

File: lib/fizzy/instantiate.cpp (hunk 3 above)
Comment (chfast) [reply lib/fizzy/instantiate.cpp:206]: The `uint32_t{}` is my proposed why of silencing the warning. I can revert these 2 changes from this PR, especially if there is a fix now in a side PR.

Comment (gumb0) [reply lib/fizzy/instantiate.cpp:206]: I would revert them if warning is disabled anyway.

**Review by gumb0 — APPROVED** [review]
